jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/audit/CaseFileDataLogDeleteBuilderImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.casemgmt.impl.audit;
+
+import java.util.Date;
+
+import org.jbpm.process.audit.JPAAuditLogService;
+import org.jbpm.process.audit.query.AbstractAuditDeleteBuilderImpl;
+import org.kie.api.runtime.CommandExecutor;
+import org.kie.internal.query.QueryParameterIdentifiers;
+
+import static org.kie.internal.query.QueryParameterIdentifiers.CASE_FILE_DATA_LOG_LASTMODIFIED;
+
+public class CaseFileDataLogDeleteBuilderImpl extends AbstractAuditDeleteBuilderImpl<CaseFileDataLogDeleteBuilder> implements CaseFileDataLogDeleteBuilder {
+
+    private static final String CASE_FILE_DATA_LOG_DELETE = "DELETE FROM CaseFileDataLog l";
+    private String caseDefId;
+
+    public CaseFileDataLogDeleteBuilderImpl(JPAAuditLogService jpaService) {
+        super(jpaService);
+        intersect();
+    }
+
+    public CaseFileDataLogDeleteBuilderImpl(CommandExecutor cmdExecutor) {
+        super(cmdExecutor);
+        intersect();
+    }
+
+    @Override
+    public CaseFileDataLogDeleteBuilderImpl dateRangeEnd(Date rangeStart) {
+        if (checkIfNull(rangeStart)) {
+            return this;
+        }
+        addRangeParameter(CASE_FILE_DATA_LOG_LASTMODIFIED, "created on date range end", ensureDateNotTimestamp(rangeStart)[0], false);
+        return this;
+    }
+
+    @Override
+    protected boolean isSubquerySupported() {
+        return true;
+    }
+
+    @Override
+    protected Subquery getSubQuery() {
+        return new Subquery("l.caseId", "select spl.correlationKey FROM ProcessInstanceLog spl", 1);
+    }
+
+    @Override
+    protected Subquery applyParameters(Subquery subquery) {
+        Subquery query = super.applyParameters(subquery);
+        query.parameter(QueryParameterIdentifiers.SUBQUERY_CASE, caseDefId);
+        return query;
+    }
+
+    @Override
+    public CaseFileDataLogDeleteBuilderImpl inCaseDefId(String caseDefId) {
```

Comment:
sry. just overlooked

Reply:
anyway using covariants return types is not really against OOP principles. Keep in mind the the client is using the interface and don't have any knowledge of the impl. Anyway just to be consistent will change it.

Reply:
OK, I have to admit that it is possibly not a good idea/or not needed just for the sake of consistency - you would have to create another interface probably where methods `logBelongsToProcessInDeployment` and `logBelongsToProcessInStatus` will be declared. I don't think we need to complicate the hierarchy just because of that... What do you think?

Reply:
We agreed to skip this step.